app/src/main/java/com/kylecorry/trail_sense/tools/qr/ui/ScanQRFragment.kt

```diff
@@ -46,9 +56,69 @@ class ScanQRFragment : BoundFragment<FragmentScanTextBinding>() {
 
     private val haptics by lazy { HapticSubsystem.getInstance(requireContext()) }
 
+    private val pickMedialoader by lazy {
+        AlertLoadingIndicator(
+            requireContext(),
+            getString(R.string.importing_image)
+        )
+    }
+
+    private val pickMedia = registerForActivityResult(ActivityResultContracts.PickVisualMedia()) { uri ->
+        uri?.let {
+            pickMedialoader.show()
+            val bitmap = BitmapFactory.decodeStream(context?.contentResolver?.openInputStream(uri))
```

Comment:
To open the bitmap (and resize it), can you use the decodeBitmapScaled method? This will be more memory efficient and match the implementation elsewhere in Trail Sense: https://github.com/kylecorry31/Trail-Sense/blob/9fc2f3bc427081266bbc8eab5f64f0c4a5242ff5/app/src/main/java/com/kylecorry/trail_sense/tools/clouds/ui/CloudResultsFragment.kt#L180

Reply:
You don't need to do the resizeExact step that the cloud results fragment does - the decodeBitmapScaled is sufficient

Reply:
I also don't recall off the top of my head if this works with external files - if it doesn't, you'll either need to copy to the temp directory like the Cloud tool does or update this to support decoding from a stream: https://github.com/kylecorry31/andromeda/blob/9b3d0ca46a71c516ef60885059f5f5b138e4906a/core/src/main/java/com/kylecorry/andromeda/core/bitmap/BitmapUtils.kt#L32